dependency/vault_pki.go

```diff
@@ -127,9 +127,12 @@ func goodFor(cert *x509.Certificate) (time.Duration, bool) {
 	lifespanDur := end.Sub(start)
 	r := rand.New(rand.NewSource(time.Now().UnixNano()))
 	lifespanMilliseconds := lifespanDur.Milliseconds()
-	// calculate the 'time the certificate should be rotated' by figuring out
-	// 87-93% of the lifespan and adding it to the start
-	rotationTime := start.Add(time.Millisecond * time.Duration(((lifespanMilliseconds*9)/10)+(lifespanMilliseconds*int64(r.Intn(6)-3))/100))
+	// calculate the 'time the certificate should be rotated' by figuring out -2%
+	// - 3% + VaultLeaseRenewalThreshold of the lifespan and adding it to the
```

Comment:
Apologies of I'm being obtuse - it seems that we are still calculating 87-93% of the VaultLeaseRenewalThresdhold (not -2% - 3% right)?

Reply:
My understanding of the math (using .80 as an example to make it easier for me)
+ float64(lifespanMilliseconds)*VaultLeaseRenewalThreshold = float64(lifespanMilliseconds) * .80
+ float64(lifespanMilliseconds*(int64(r.Intn(6)-3)/100.0)) = float64(lifespanMilliseconds) * [-.03, .03]

`float64(lifespanMilliseconds) * .80 +  float64(lifespanMilliseconds) * [-.03, .03]` - which would come out to .77 - .83 of lifespanMilliseconds which is -3, +3 of VaultRenewalThreshold

Reply:
Ah yes sorry makes sense... It might make sense to update that comment to show "-3, +3" instead of "-3, -3"

Reply:
I double checked and r.Intn() is actually [0, n) so it would be [-3, 2]. I updated the code